cmd/template/nodepool/provider/aws.go

```diff
@@ -64,3 +149,59 @@ func WriteAWSTemplate(out io.Writer, config NodePoolCRsConfig) error {
 
 	return nil
 }
+
+func getCAPANodepoolTemplate(config NodePoolCRsConfig) (client.Template, error) {
+	var err error
+
+	c, err := client.New("")
+	if err != nil {
+		return nil, err
+	}
+
+	templateOptions := client.GetClusterTemplateOptions{
+		ClusterName:       config.ClusterID,
+		TargetNamespace:   config.Owner,
+		KubernetesVersion: "v1.19.9",
+		ProviderRepositorySource: &client.ProviderRepositorySourceOptions{
+			InfrastructureProvider: "aws:v0.6.6",
+			Flavor:                 "machinepool",
+		},
+	}
+	os.Setenv("AWS_SUBNET", "")
+	os.Setenv("AWS_CONTROL_PLANE_MACHINE_TYPE", "")
+	os.Setenv("AWS_REGION", "")
+	os.Setenv("AWS_SSH_KEY_NAME", "")
```

Comment:
For some reason the client DEMANDS these envs to be set. Left empty for defaulting.

Reply:
I think we should check if these are already set before setting them to an empty string and also [unset](https://golang.org/pkg/os/#Unsetenv) these once no longer required.

Reply:
Yeah, +1 on that

Reply:
I agree with unsetting them! Will add that.
However, if we check if they are already set and leave that value, it could lead to unexpected different outcomes. I would prefer to leave them as empty strings and transparently default in kyverno.

Reply:
makes sense I guess

Reply:
Was this the right way to solve this though? In environments having these set, we should revert them to the value before, so something like

```go
if os.Getenv("AWS_SUBNET") != "" {
  prevEnvAWSSubnet := os.Getenv("AWS_SUBNET")
  defer os.Setenv("AWS_SUBNET", prevEnvAWSSubnet)
} else {
  defer os.Unsetenv("AWS_SUBNET")
}
```

Reply:
True, good point. Follow up: https://github.com/giantswarm/kubectl-gs/pull/390